embassy-traits/src/qei.rs

```diff
@@ -0,0 +1,13 @@
+use core::future::Future;
+use core::pin::Pin;
+use embedded_hal::Direction;
+
+pub trait Qei {
```

Comment:
The name is not very descriptive. Maybe QuadratureDecoder, or WaitForRotation?

(Why does stm32 call this QEI for Quadrature Encoder? It *decodes* a quadrature-encoded signal. NRF calls it QDEC)

Reply:
it's called an encoder because the device that it measures is called an encoder 

https://www.mouser.com/ProductDetail/CUI-Devices/C14D16N-C3P?qs=vEM7xhTegWjy3URF2TfWBw%3D%3D